herbert_core/utilities/maths/noisify.m

```diff
@@ -25,32 +34,44 @@
         for i=1:numel(y)
             yout(i)=randpoisson(abs(y(i)));
         end
-        eout=abs(y);  % the inpuy y is the mean and variance of the Poisson distribution
+        eout=abs(y);  % the input y is the mean and variance of the Poisson distribution
     else
         error('Unrecognised random sampling distribution')
     end
-else
-    if nargin==3
+else % 3rd arg did not exist or was not a distribution name
+    default_fac = 0.1;
+    if nargin>=3 % 3 args for fac or 4 args for fac and maxval
         if isnumeric(varargin{1}) && isscalar(varargin{1})
             fac = varargin{1};
+            if fac < 0.0 % default option if 4th arg is needed
+                fac = default_fac;
+            end
         else
             error('Noise as fraction of peak signal must be real scalar')
         end
     else
-        fac = 0.1;
+        fac = default_fac;
+    end
+    if nargin==4 % means the maxval arg has been used
+        ymax = varargin{2};             % use max value previously extracted
+    else
+        ymax = max(abs(y(:)));          % find maximum magnitude of y for arbitrary dimensions
     end
-    ymax = max(abs(y(:)));          % find maximum magnitude of y for arbitrary dimensions
     dy=(fac*ymax)*randn(size(y));   % st. dev. of randn is sigma=1
     yout=y+dy;
     eout=ones(size(y))*(fac*ymax)^2;
 end
 
-if exist('e','var') && isempty(e)
+% this code now outputs a variance to be consistent between the use of eout
```

Comment:
I don't think these two comments are adding any value for future developers:
- to anyone reading the code as it is now you have defined e as a variance at the top; so they wouldn't expect or think it was an stddev
- the missing `~` was a bug you've fixed. We don't need to document that beyond the git log

Reply:
Cleaned up with commit 4